src/module/rules/rule-element/special-resource.ts

```diff
@@ -116,15 +117,16 @@ class SpecialResourceRuleElement extends RuleElementPF2e<SpecialResourceSchema>
 
         // Keep a record of the resource for blacklisting and redirection purposes
         // Also prepare a basic version for active effect like modification
+        const key = sluggify(this.slug, { camel: "dromedary" });
         this.max = Number(this.resolveValue(this.max));
-        if (!(this.slug in this.actor.synthetics.resources)) {
-            this.actor.synthetics.resources[this.slug] = this;
-            this.actor.system.resources[this.slug] = fu.mergeObject(this.actor.system.resources[this.slug] ?? {}, {
+        if (!(key in this.actor.synthetics.resources)) {
+            this.actor.synthetics.resources[key] = this;
+            this.actor.system.resources[key] = fu.mergeObject(this.actor.system.resources[key] ?? {}, {
                 value: 0,
                 max: this.max,
             });
```

Comment:
Sorry, I forgot: what is the purpose of setting it in both synthetics and system data?

Reply:
The system data is the true prepared data (and allows parity with true resources like hero points), and the synthetic allows us to perform updates.